store/v2/commitment/store.go

```diff
@@ -28,9 +30,10 @@ var (
 	_ store.UpgradeableStore      = (*CommitStore)(nil)
 	_ snapshots.CommitSnapshotter = (*CommitStore)(nil)
 	_ store.PausablePruner        = (*CommitStore)(nil)
-	_ store.KVStoreGetter         = (*CommitStore)(nil)
 )
 
+type MountTreeFn func(storeKey string) (Tree, error)
```

Comment:
needs a godoc if to be included

Reply:
is there an example usage of this in tests? trying to wrap my head around how it will work in GetProof.  would multiple different SC still be supported? the current has a homogenous `mounTreeFn` for heterogeneous `map[string]Tree`